src/main/java/edu/cornell/kfs/coa/batch/service/impl/CopyLegacyAccountAttachmentsServiceImpl.java

```diff
@@ -0,0 +1,237 @@
+        final CopyLegacyAccountAttachmentsService proxiedService =
+                getThisServiceAsProxyToProcessAttachmentsInNewTransactions();
+        int numCopiedAttachments = 0;
+        int numCopyFailures = 0;
+
+        for (final LegacyAccountAttachment attachmentToCopy : attachmentsToCopy) {
+            if (proxiedService.copyLegacyAccountAttachmentToKfs(attachmentToCopy)) {
+                numCopiedAttachments++;
+            } else {
+                numCopyFailures++;
+                if (numCopyFailures > maxFailuresAllowed) {
+                    LOG.error("copyLegacyAccountAttachmentsToKfs, More than {} attachments failed to copy; aborting",
+                            maxFailuresAllowed);
+                    return false;
+                }
+            }
+        }
+
+        LOG.info("====================");
+        LOG.info("copyLegacyAccountAttachmentsToKfs, Successfully copied {} legacy account attachments into KFS",
+                numCopiedAttachments);
+        LOG.info("copyLegacyAccountAttachmentsToKfs, {} legacy account attachments failed copying into KFS",
+                numCopyFailures);
+        LOG.info("====================");
+
+        return true;
+    }
+
+    private CopyLegacyAccountAttachmentsService getThisServiceAsProxyToProcessAttachmentsInNewTransactions() {
+        return SpringContext.getBean(CopyLegacyAccountAttachmentsService.class);
+    }
+
+    @Transactional(propagation = Propagation.REQUIRES_NEW)
+    @Override
+    public boolean copyLegacyAccountAttachmentToKfs(final LegacyAccountAttachment legacyAccountAttachment) {
+        try {
+            LOG.debug("copyLegacyAccountAttachmentToKfs, Copying attachment: {}", legacyAccountAttachment);
+            final Account account = getKfsAccountForAttachment(legacyAccountAttachment);
+            final AccountingXmlDocumentBackupLink backupLink = createBackupLinkForAttachmentDownload(
+                    legacyAccountAttachment);
+            final Attachment attachment = accountingXmlDocumentDownloadAttachmentService
+                    .createAttachmentFromBackupLink(account, backupLink);
+            saveAttachmentToKfs(attachment, account, backupLink.getDescription());
+            LOG.info("copyLegacyAccountAttachmentToKfs, Copied attachment with ID '{}' targeting account '{}-{}'",
+                    legacyAccountAttachment.getId(), legacyAccountAttachment.getKfsChartCode(),
+                    legacyAccountAttachment.getKfsAccountNumber());
+            copyLegacyAccountAttachmentsDao.markLegacyAccountAttachmentAsCopied(legacyAccountAttachment);
+            return true;
+        } catch (final Exception e) {
+            LOG.error("copyLegacyAccountAttachmentToKfs, Failed to copy attachment: {}", legacyAccountAttachment, e);
+            copyLegacyAccountAttachmentsDao.recordCopyingErrorForLegacyAccountAttachment(
+                    legacyAccountAttachment, e.getMessage());
+            return false;
```

Comment:
Given the discovery that you made on KFSPTS-33040, I agree that it would be best for this method to re-throw the exception (or a wrapped exception) so that the rollback will happen accordingly. I'll adjust this part to do so, and I'll rework the logic so that the error-recording step will occur in bulk within the parent transaction (or a new standalone sub-transaction).